viewing_party/party.py

```diff
@@ -0,0 +1,122 @@
+def create_movie(movie_title, genre, rating):
+    if movie_title and genre and rating:
+        new_movie = {
+            "title": movie_title,
+            "genre": genre,
+            "rating": rating
+        }
+        return new_movie
+    
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_to_watch):
+    for movie_data in user_data["watchlist"]:
+        if movie_data["title"] == movie_to_watch:
+            user_data["watched"].append(movie_data)
+            user_data["watchlist"].remove(movie_data)
+    
+    return user_data
+
+def get_watched_avg_rating(user_data):
+    average_data = []
+
+    for movie_data in user_data["watched"]:
+        average_data.append(movie_data["rating"])
+
+    if not average_data:
+        return 0
+
+    return sum(average_data)/len(average_data)
+        
+def get_most_watched_genre(user_data):
+    most_watched = {}
+
+    for movie_data in user_data["watched"]:
+        if movie_data["genre"] not in most_watched:
+            most_watched[movie_data["genre"]] = 1
+        else:
+            most_watched[movie_data["genre"]] += 1
+
+    if most_watched:
+        return max(most_watched, key=most_watched.get)
+
+    return None
+    
+def get_unique_watched(user_data):
+    unique_watched = []
+    unique_watched += user_data["watched"]
+
+    for friend_data in user_data["friends"]:
+        for movie_data in friend_data["watched"]:
+            if movie_data in unique_watched:
+                unique_watched.remove(movie_data)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    unique_watched = []
+
+    for friend_data in user_data["friends"]:
+        unique_watched += friend_data["watched"]
+    
+    for movie_data in user_data["watched"]:
+        if movie_data in unique_watched:
+            unique_watched.remove(movie_data)
+
+    real_unique =[]
+    [real_unique.append(x) for x in unique_watched if x not in real_unique]
```

Comment:
I generally do not recommend using list comprehensions for side effects.  I would suggest writing this out as a full for loop.